src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -425,6 +425,27 @@ public class ConfigurationTests
             }
         }";
 
+        internal const string GRAPHQL_SCHEMA_WITH_CYCLE = @"
+type Character {
+    id : ID,
+    name : String,
+    star: Planet,
+    moons: [Moon],
+}
+
+type Planet @model(name:""Planet"") {
+    id : ID!,
+    name : String,
+    character: Character
+}
+
+type Moon {
+    id : ID,
+    name : String,
+    details : String,
+    character: Character
+}
```

Comment:
because Character.moons seems to be an array type (`[Moon]`), does your code work differently if Moon.character were processed last vs. Character.moons?

Reply:
any other examples of circular reference or does this cover all circumstances?

Reply:
yes, it should cover all circumstances. Let me know if want me to cover any particular scenario. I added one more test to cover array and no-array object in circular reference separately.